File: qz/u2/santi/Exam U2/71-100/ExamSpeakers/src/main/java/ec/edu/espe/examspeakers/model/Speaker.java

package ec.edu.espe.examspeakers.model;

/**
 *
 * @author Sjean
 */
public class Speaker {
    private String id;
    private String brand;
    private double power;
    private double frequency;

    public Speaker(String id, String brand, double power, double frequency) {
        this.id = id;
        this.brand = brand;
        this.power = power;
        this.frequency = frequency;
    }

    // Getters and setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getBrand() { return brand; }
    public void setBrand(String brand) { this.brand = brand; }
    public double getPower() { return power; }
    public void setPower(double power) { this.power = power; }
    public double getFrequency() { return frequency; }
    public void setFrequency(double frequency) { this.frequency = frequency; }

    public double calculateQualityIndex() {
        return power * frequency;
    }
}

File: qz/u3/santi/qz1&2/Quizz_Speakers/Quizz_Speakers/model/speaker.py
class Speaker:
    def __init__(self, speaker_id: str, brand: str, power: float, frequency: float):
        self.id = speaker_id
        self.brand = brand
        self.power = power
        self.frequency = frequency

    def calculate_quality_index(self) -> float:
        return self.power * self.frequency

    # Getters and setters
    def get_id(self):
        return self.id

    def set_id(self, speaker_id):
        self.id = speaker_id

    def get_brand(self):
        return self.brand

    def set_brand(self, brand):
        self.brand = brand

    def get_power(self):
        return self.power

    def set_power(self, power):
        self.power = power

    def get_frequency(self):
        return self.frequency

    def set_frequency(self, frequency):
        self.frequency = frequency
